### PR TITLE
Get Link and Joints scoped names

### PR DESCRIPTION
**Pull request by diegoferigo**

### CODE REVIEW

File: cpp/scenario/gazebo/include/scenario/gazebo/Link.h [hunk 1]
```diff
@@ -68,8 +68,15 @@ class scenario::gazebo::Link
     // Link Core
     // =========
 
-    std::string name() const;
+    /**
+     * Get the name of the link.
+     *
+     * @param scoped Return the scoped link name.
```
Comment (traversaro) [inline cpp/scenario/gazebo/include/scenario/gazebo/Link.h:74]: The docs say it return the scoped name, but the default value for scoped is `false` .

Comment (diegoferigo) [reply cpp/scenario/gazebo/include/scenario/gazebo/Link.h:74]: Good catch! Thanks

**Comment by traversaro** [comment]
Related issue: https://github.com/ignitionrobotics/design/issues/3 .

**Comment by diegoferigo** [comment]
> There is any documentation in Ign Gazebo on what a "scoped" name is? I remember in the past some ambiguous definitions especially with nested models.

Mmh I didn't dig that much. I know that in Gazebo scoped names are a thing, and for us they were a requirement for the contact system. In fact, in this case, you want to identify two links in contact that could belong to different models. Using scoped names was the first solution that came in my mind.

To be honest I don't know how nested models are handled, and I never though about them. Just to be sure, a file like [`CartPole.world`](https://github.com/robotology/gym-ignition/blob/4def996c4226e8ffec5ff882b6efff54dce3549f/gym_ignition_data/worlds/CartPole.world) would generate a nested model? If yes, then the ECM abstracts it away and flattens out the hierarchy.

> Personal comment: boolean parameters are not really readable in C++ ( https://abseil.io/tips/94 ), in this case two different methods name() and scopedName() may be more readable.

Thanks for the suggestion. I'm trying to use as much as possible similar solutions because the APIs are already quite big, particularly for joints and models, and merging simple methods like this one is quite helpful to clear the clutter. Interestingly in the abseil tip they also recommend the pattern I typically use:

https://github.com/robotology/gym-ignition/blob/79fee68a15d0bf69c7222c04b75c32dcefa9d462/cpp/scenario/gazebo/src/GazeboSimulator.cpp#L669-L670

A similar Python example:

 https://github.com/robotology/gym-ignition/blob/79fee68a15d0bf69c7222c04b75c32dcefa9d462/tests/test_scenario/test_gazebo_simulator.py#L60

**Comment by diegoferigo** [comment]
> Related issue: ignitionrobotics/design#3 .

Good catch! I missed it :)

File: cpp/scenario/gazebo/include/scenario/gazebo/Link.h (hunk 1 above)
Comment (diegoferigo) [reply cpp/scenario/gazebo/include/scenario/gazebo/Link.h]: Addressed in https://github.com/robotology/gym-ignition/compare/c3162c59ecfd0fa84573201c16f33050ed4eb383..2fbe4410c7caf590d78e5008106321c829d7acd8.

**Pull request merged**